src/main/java/org/wirvsvirus/locoronando/location/GeocodingServiceImpl.java

```diff
@@ -0,0 +1,57 @@
+package org.wirvsvirus.locoronando.location;
+
+import java.util.Optional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+import org.wirvsvirus.locoronando.dealer.Address;
+
+@Service
+public class GeocodingServiceImpl implements GeocodingService {
```

Comment:
Implement the methods directly in the GeordingService instead of adding a class. You don't want to implement the interface more than once, do you?

Reply:
If we need to switch the external service, we might want to...